incubator/binding-openapi/src/main/java/io/aklivity/zilla/runtime/binding/openapi/internal/streams/OpenapiClientFactory.java

```diff
@@ -0,0 +1,1059 @@
+        }
+
+        private void onOpenapiReset(
+            ResetFW reset)
+        {
+            final long sequence = reset.sequence();
+            final long acknowledge = reset.acknowledge();
+            final int maximum = reset.maximum();
+            final long traceId = reset.traceId();
+
+            assert acknowledge <= sequence;
+            assert sequence <= replySeq;
+            assert acknowledge >= replyAck;
+            assert maximum >= replyMax;
+
+            replyAck = acknowledge;
+            replyMax = maximum;
+            state = OpenapiState.closeReply(state);
+
+            assert replyAck <= replySeq;
+
+            cleanup(traceId);
+        }
+
+        private void onOpenapiWindow(
+            WindowFW window)
+        {
+            final long sequence = window.sequence();
+            final long acknowledge = window.acknowledge();
+            final int maximum = window.maximum();
+            final long traceId = window.traceId();
+            final long budgetId = window.budgetId();
+            final int padding = window.padding();
+            final int capabilities = window.capabilities();
+
+            assert acknowledge <= sequence;
+            assert sequence <= replySeq;
+            assert acknowledge >= replyAck;
+            assert maximum >= replyMax;
+
+            replyAck = acknowledge;
+            replyMax = maximum;
+            replyBud = budgetId;
+            replyPad = padding;
+            replyCap = capabilities;
+            state = OpenapiState.closingReply(state);
+
+            assert replyAck <= replySeq;
+
+            http.doHttpWindow(traceId, acknowledge, budgetId, padding);
+        }
+
+        private void doOpenapiBegin(
+            long traceId,
+            OctetsFW extension)
+        {
+            state = OpenapiState.openingReply(state);
+
+            final OpenapiBeginExFW openBeginEx = openBeginExRW
+                .wrap(extBuffer, 0, extBuffer.capacity())
+                .typeId(openapiTypeId)
+                .operationId(operationId)
+                .extension(extension)
+                .build();
+
+            doBegin(sender, originId, routedId, replyId, replySeq, replyAck, replyMax,
+                traceId, authorization, affinity, openBeginEx);
+        }
+
+        private void doOpenapiData(
+            long traceId,
+            int flag,
+            int reserved,
+            OctetsFW payload,
+            Flyweight extension)
+        {
+
+            doData(sender, originId, routedId, replyId, replySeq, replyAck, replyMax,
+                    traceId, authorization, replyBudgetId, flag, reserved, payload, extension);
+
+            replySeq += reserved;
+        }
+
+        private void doOpenapiFlush(
+            long traceId,
+            OctetsFW extension)
+        {
+            doFlush(sender, originId, routedId, replyId, replySeq, replyAck, replyMax,
+                traceId, authorization, replyBudgetId, 0, extension);
```

Comment:
We need to propagate `reserved` on flush too, also keep `replySeq` updated similar to `doOpenapiData`..